tests/testthat/test-pca-serde.R

```diff
@@ -0,0 +1,51 @@
+test_that("PCA models can be serialized and unserialized correctly", {
+  models <- list()
+
+  for (whiten in c(FALSE, TRUE)) {
+    for (transform_input in c(FALSE, TRUE)) {
+      model <- cuda_ml_pca(
+        iris[1:4], n_components = 3, whiten = whiten,
+        transform_input = transform_input
+      )
+      models <- append(models, list(model))
+    }
+  }
+
+  for (model in models) {
+    expected_inv_transform <- cuda_ml_inverse_transform(model, iris[1:4])
+    actual_inv_transform <- callr::r(
+      function(model_state, expected_components, expected_expl_var,
+               expected_expl_var_ratio, expected_sg_vals, expected_m,
+               expected_tf_data) {
+        library(cuda.ml)
+        library(testthat)
+
+        expect_true(has_libcuml())
+
+        model <- cuda_ml_unserialize(model_state)
+
+        expect_equal(model$components, expected_components)
+        expect_equal(model$explained_variance, expected_expl_var)
+        expect_equal(
+          model$explained_variance_ratio, expected_expl_var_ratio
+        )
+        expect_equal(model$singular_values, expected_sg_vals)
+        expect_equal(model$mean, expected_m)
+        expect_equal(model$transformed_data, expected_tf_data)
```

Comment:
Do you anticipate this list of attributes accessible under `model$` to grow with future cuml releases? Will we get those additional attributes "for free"? If yes, would it make sense to just iterate through all elements of `as.list(model)` (or something equivalent?)

Flip side, are there any attributes you don't expect check to survive saving/restoring? If yes, maybe a `expect_null` or `expect_error`?

Reply:
I don't expect additional attributes here in future versions of cuML.